digestparser/json_output.py

```diff
@@ -0,0 +1,183 @@
+"build JSON output from digest content"
+import os
+import time
+import copy
+from collections import OrderedDict
+import requests
+from elifetools.utils import copy_attribute
+from digestparser.utils import msid_from_doi
+from digestparser.jats import (parse_jats_file, parse_jats_digest, parse_jats_pub_date,
+                               parse_jats_subjects, xml_to_html)
+from digestparser.build import build_digest
+
+
+def iiif_server_info(info_url):
+    "get the image info from the IIIF server"
+    info = {}
+    try:
+        response = requests.get(info_url)
+        info = response.json()
+    except:
+        # could be any error right now
+        pass
+    return info
+
+
+def image_info(msid, file_name, digest_config):
+    "get image info from the IIIF server"
+    if not msid or not file_name or not digest_config:
+        return {}
+    info_url = digest_config.get('iiif_info_url').format(msid=msid, file_name=file_name)
+    return iiif_server_info(info_url)
+
+
+def image_attribution(credit, image_license):
+    "concatenate an image attribution"
+    attribution = []
+    attribution_line = ', '.join([part for part in [credit, image_license] if part])
+    attribution.append(attribution_line)
+    return attribution
+
+
+def image_uri(msid, file_name, digest_config):
+    "uri of the image file as defined in the settings"
+    return digest_config.get('iiif_image_uri').format(msid=msid, file_name=file_name)
+
+
+def image_source(msid, file_name, digest_config):
+    "source of the iiif image as defined in the settings"
+    source = OrderedDict()
+    source['mediaType'] = "image/jpeg"
+    source['uri'] = digest_config.get('iiif_image_source_uri').format(msid=msid,
+                                                                      file_name=file_name)
+    source['filename'] = file_name
+    return source
+
+
+def image_size(info):
+    "size of the iiif image from the info"
+    size = OrderedDict()
+    # get the size from the json
+    size['width'] = info.get('width')
```

Comment:
I looked at how the image data is gathered and presented, also in combination with a comment related to image below. If there is a small bit of image data missing, such as the width and height, it is an option to completely remove all image JSON. 

However, since this is a library, I like it to add as much data about an image as possible. If it produces at least partial JSON, this library shouldn't be concerned with whether it is valid against a particular schema. It is also possible the image `size` will be added later by a separate operation in another workflow. 

It is in keeping with the test scenarios where I've illustrated how you can produce a number of partial JSON outputs depending on how much input data is available.

Another change could be for `build_json()` to accept `image_info`, and if you supplied it (which contains the width and height in a dict) then this library does not need to issue a IIIF request. It already accepts `related` data, so this library did not also need to understand how to data from Lax for related articles.

A long comment, sorry for that. In summary, if the image info is missing then `size` is omitted from the JSON output. It will still be invalid against certain schemas, but it will never have `width: None`.